api/allocations.go

```diff
@@ -4,6 +4,8 @@ import (
 	"fmt"
 	"sort"
 	"time"
+
+	"github.com/hashicorp/nomad/nomad/structs"
```

Comment:
The API package shouldn't reference the internal nomad/structs package - doing so leaks many internal packages and transitive dependencies to api clients and make dependency management more complicated, specially for go mod users.

Can you copy the relevant structs here?